doc/sphinxext/gallery_generator.py

```diff
@@ -59,136 +59,33 @@
-        filter:alpha(opacity=100); /* For IE8 and earlier */
-    }}
-
-    span.figure-label {{
-        position: absolute;
-        display: inline;
-        left: 0;
-        width: 170px;
-        height: 170px;
-        background: #000;
-        color: #fff;
-        visibility: hidden;
-        opacity: 0;
-        z-index: 100;
-    }}
-
-    .figure p {{
-        position: absolute;
-        top: 45%;
-        width: 170px;
-        font-size: 110%;
-    }}
-
-    .figure:hover span {{
-        visibility: visible;
-        opacity: .4;
-    }}
-
-    .caption {{
-        position: absolute;
-        width: 180px;
-        top: 170px;
-        text-align: center !important;
-    }}
-
-    .figure .gallery-figure-title p {{
-        position: relative;
-        top: 170px;
-        color: black;
-        visibility: visible;
-        text-align: center !important;
-        line-height: normal;
-    }}
-    .figure .gallery-figure-title span {{
-        top: 170px;
-        position: relative;
-        visibility: visible;
-    }}
-    </style>
-
-.. _{sphinx_tag}:
-
 Example gallery
 ===============
+{toctree}
+{contents}
+"""
 
-{toctrees_contents}
+TOCTREE_START = """
+.. toctree::
+   :hidden:
 
 """
 
-CONTENTS_ENTRY_TEMPLATE = (
-    ".. raw:: html\n\n"
-    "    <div class='figure align-center'>\n"
-    "    <a href=./{htmlfilename}>\n"
-    "    <img src=../_static/{thumbfilename}>\n"
-    "    <span class='figure-label'>\n"
-    "    <p>{sphinx_tag}</p>\n"
-    "    </span>\n"
-    '    <span class="gallery-figure-title">\n'
-    "      <p>{title}</p>\n"
-    "    </span>\n"
-    "    </a>\n"
-    "    </div>\n\n"
-    "\n\n"
-    ""
-)
-
-
-def create_thumbnail(infile, thumbfile, width=275, height=275, cx=0.5, cy=0.5, border=4):
-
-    im = image.imread(infile)
-    rows, cols = im.shape[:2]
-    size = min(rows, cols)
-    if size == cols:
-        xslice = slice(0, size)
-        ymin = min(max(0, int(cy * rows - size // 2)), rows - size)
-        yslice = slice(ymin, ymin + size)
-    else:
-        yslice = slice(0, size)
-        xmin = min(max(0, int(cx * cols - size // 2)), cols - size)
-        xslice = slice(xmin, xmin + size)
-    thumb = im[yslice, xslice]
-    thumb[:border, :, :3] = thumb[-border:, :, :3] = 0
-    thumb[:, :border, :3] = thumb[:, -border:, :3] = 0
-
-    dpi = 100
-    fig = plt.figure(figsize=(width / dpi, height / dpi), dpi=dpi, constrained_layout=True)
-
-    ax = fig.add_axes([0, 0, 1, 1], aspect="auto", frameon=False, xticks=[], yticks=[])
-    ax.imshow(thumb, aspect="auto", resample=True, interpolation="bilinear")
-    fig.savefig(thumbfile, dpi=dpi)
-    plt.close(fig)
+CONTENTS_START = """
+.. grid:: 3
+   :gutter: 2 2 3 3
+"""
+
+CONTENTS_ENTRY_TEMPLATE = """
+   .. grid-item-card:: 
+      :link: ./{htmlfilename}
+      :img-top: ./matplotlib/_images/{imgfilename}
+      :text-align: center
+      :shadow: none
+      :class-card: example-gallery
+
+      {title}
+"""
```

Comment:
> FYI: Keeping the ratio intact while making sure images take up the same amount of real estate per card was handled in the CSS (.example-gallery img.sd-card-img-top and .example-gallery .sd-card-body). This isn't really the normal way to resize images but it works. The more normal would be to create a <div> container with fixed height and have the img fill up the container, but I don't think Sphinx has this flexibility.

I think this is possible. Cards in sphinx-design can add content in up to 3 separated divs (header, body, footer, the default being to use a single body div). I think these could be used for that. It is also possible to define divs (and their classes) as an rst directive, but that would be a bit more cumbersome. It looks like the issue here is that using img-top does not place the image inside a specific div. I think something like this would work:

```suggestion
CONTENTS_ENTRY_TEMPLATE = """
   .. grid-item-card:: 
      :link: ./{htmlfilename}
      :text-align: center
      :shadow: none
      :class-card: example-gallery
      :class-header: specific div size fixing class here
      
      .. image:: ./matplotlib/_images/{imgfilename}
         :alt: alt text would go here (optional for now, could be the description we discussed)
      
      ^^^

      {title}
"""
```

classes passed in `class-card` affect the outer card div, `class-header` classes affect only the content above the `^^^` and it is also possible to use `class-body` with classes that affect only the content below `^^^`.

Reply:
okay THANK YOU! This works better and now I can center align things easier with more legible CSS :)

Reply:
Glad I could help. As I said, I don't know much about css and no js. But I do know a lot of sphinx (using either rst or markdown) 😊